### ISSUE TITLE
unblacklist rust recipes

### ISSUE THREAD

**Issue by dlaehnemann** [bioconda/bioconda-recipes#9266]
This is a tracking issue for progress with unblacklisting the two remaining rust recipes in the build blacklist in issue #7712, I.e. these lines (plus `rust-bio-tools`, which should probably need another look, as well, and `spalter` and `prosolo` can probably do with the same simultaneous changes):
https://github.com/bioconda/bioconda-recipes/blob/642ad05c02b9d23441108e09fee795390844aa90/build-fail-blacklist#L37-L39

The remaining issue boils down to some problem with the combination of `conda build` version `2.` and the `rust` compiler `cargo`, so the easiest is probably to just wait for the coming `bioconda-utils` update to `3.x` `conda build`:
https://github.com/bioconda/bioconda-utils/pull/293

The below infos are to document the previous attempt, but hopefully most of these workarounds will not be necessary any more. There is a branch I created with some attempts at getting these recipes unblacklisted, but it should probably be retired in favor of a clean new branch and should only cherry-pick relevant commits, if any:
https://github.com/bioconda/bioconda-recipes/tree/unblacklist_rust_recipes

The commits there incorporate the workarounds identified when unblacklisting prosolo (#7973) and further recommendations given for the new recipe for spalter that's on the way (#9062). But none of these got all the way and eventually we end up with an awkward interaction between the older version of `conda build` currently used (`2.1.18`) and `cargo`'s checking that certain paths are absolute paths:
```
thread 'main' panicked at 'assertion failed: src_path.is_absolute()', tools/cargo/src/cargo/core/manifest.rs:438:9
```

It goes away when I locally use `conda build` version `3.4.2`, hence the above suggestions to wait for the `bioconda-utils` update.

It does not go away when using older cargo versions, but the next future `rust`/`cargo` version (`1.27`) should include a change that would at least give a more informative panic message, should more debugging be needed in the future:
https://github.com/rust-lang/cargo/commit/dfdb9aab12c89257fbd529719265ac65609a55a3

**Comment by bgruening** [comment]
@dlaehnemann any updates on this one?

**Comment by dlaehnemann** [comment]
Thanks for the reminder, I'll try to have a look in the coming days!

**Comment by dlaehnemann** [comment]
`rust-overlaps` was already fixed by @luizirber in PR #9670 where he references this issue.

`prosic` has been superseded by [`varlociraptor`](https://bioconda.github.io/recipes/varlociraptor/README.html). But I would say a final unblacklisted, building version of prosic would be nice, and @luizirber also suggested a way forward by just tagging a newer working version and releasing a it with the bug fix version number increased:
* https://github.com/varlociraptor/prosic2/issues/1
* PR #9674

@johanneskoester: Would this still be an option? Just releasing a prosic2 `2.1.1`?

**Comment by dlaehnemann** [comment]
There now is a merged `prosic2 2.1.2` and `prosic` is unblacklisted. See PR #18197
